### PR TITLE
[FIX] form: always redirect to become_cooperator

### LINKED ISSUE #295: As an existing user, I can't subscribe to a share in the online form
- odoo finds an existing user
- click on "do you have an account" and login

=> some (required) fields are not filled
Screenshot 2022-04-02 at 19 13 55

### PR DESCRIPTION
**Pull request by victor-champonnois**
fix https://github.com/coopiteasy/vertical-cooperative/issues/295

Not based on 12.0-komunigi since it is not related to the komunigi refactoring.

This is a dirty fix to a structural issue in the controller (having two routes for the same template). 
Conceived with @remytms

### CODE REVIEW

File: easy_my_coop_website/controllers/main.py [hunk 1]
```diff
@@ -249,6 +252,8 @@ def validation(  # noqa: C901 (method too complex)
         sub_req_obj = request.env["subscription.request"]
 
         redirect = "easy_my_coop_website.becomecooperator"
+        # redirect url to fall back on become coopererator in template redirection
+        values["redirect_url"] = request.httprequest.host_url + "become_cooperator"
```
Comment (robinkeunen) [inline easy_my_coop_website/controllers/main.py]: I think it's safer to use `from urllib.parse import urljoin` to join url : it makes sure the `/` are present and not doubled among other things.

**Review by robinkeunen — CHANGES_REQUESTED** [review]
Another review from @remytms would be safer.

File: easy_my_coop_website/controllers/main.py (hunk 1 above)
Comment (victor-champonnois) [reply easy_my_coop_website/controllers/main.py]: Yes I agree, it's fixed :)

**Review by robinkeunen — APPROVED** [review]

**Review by carmenbianca — APPROVED** [review]

**Pull request merged**